File: C++/Algorithms/Tree Algorithms/LowestCommonAncestor.cpp
 TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        
       if we have one less than one greater than root the obvious thats the ancestor so this way we are finding ancestor
if(root->val>p->val&&root->val>q->val)return lowestCommonAncestor(root->left,  p, q);
 else if(root->val<p->val&&root->val<q->val)return lowestCommonAncestor(root->right,p,q);
  return root;  
    }

File: Java/Algorithms/BinarySearchTree/lowestcommonancestor.java

static TreeNode lca(TreeNode root,int d1,int d2){
    if(root.val> d1 && root.val>d2){
        return lca(root.left,d1,d2);
    }
    else if(root.val <d1 && root.val<d2){
        return lca(root.right,d1,d2);
    }
    return root;
}